src/goMain.ts

```diff
@@ -57,8 +57,16 @@ export function activate(ctx: vscode.ExtensionContext): void {
 
 	updateGoPathGoRootFromConfig().then(() => {
 		const updateToolsCmdText = 'Update tools';
-		const prevGoroot = ctx.globalState.get('goroot');
-		const currentGoroot = process.env['GOROOT'];
+		interface GoInfo {
+			goroot: string;
+			version: string;
+		}
+		const toolsGopath = getToolsGopath() || getCurrentGoPath();
+		let toolsGoInfo: { [id: string]: GoInfo; } = ctx.globalState.get('toolsGoInfo') || {};
+		if (!toolsGoInfo[toolsGopath]) toolsGoInfo[toolsGopath] = { goroot: null, version: null };
+		const prevGoroot = toolsGoInfo[toolsGopath].goroot;
```

Comment:
@FiloSottile Is this to cater to the case where you have different goroots as well as toolgopaths for different workspaces. Each of these toolGopaths have the Go tools compiled using the right version of go and so recompiling is not needed?